serde/src/de/format.rs

```diff
@@ -1,19 +1,19 @@
 use lib::fmt::{self, Write};
 use lib::str;
 
-pub struct Buf<'a> {
+pub(super) struct Buf<'a> {
     bytes: &'a mut [u8],
     offset: usize,
 }
 
 impl<'a> Buf<'a> {
-    pub fn new(bytes: &'a mut [u8]) -> Self {
+    pub(super) fn new(bytes: &'a mut [u8]) -> Self {
         Buf { bytes, offset: 0 }
     }
 
-    pub fn as_str(&self) -> &str {
+    pub(super) unsafe fn as_str(&self) -> &str {
```

Comment:
I don't see why this needs to be unsafe.

Reply:
> - It changes `Buf::as_str()` to be unsafe, since it relies on the caller to only write UTF-8 into the buffer.

The API enforces that the caller only wrote UTF-8 into the buffer.

Reply:
Hi dtolnay! Oh you are totally right. I’ll fix it.